indigo/indigo/src/main/scala/indigo/platform/events/WorldEvents.scala

```diff
@@ -111,55 +140,252 @@ final class WorldEvents:
+          globalEventStream.pushGlobalEvent(
+            MouseEvent.Enter(
+              position,
+              buttons,
+              e.altKey,
+              e.ctrlKey,
+              e.metaKey,
+              e.shiftKey,
+              movementPosition
+            )
+          )
+        }
       },
       onPointerLeave = { e =>
-        val position = e.position(magnification, canvas)
+        val position         = e.position(magnification, canvas)
+        val buttons          = e.indigoButtons
+        val movementPosition = e.movementPosition(magnification)
+        val pointerType      = e.indigoPointerType
 
         globalEventStream.pushGlobalEvent(
-          PointerLeave(position, PointerId(e.pointerId), Buttons(e.buttons), e.isPrimary)
+          PointerLeave(
+            position,
+            buttons,
+            e.altKey,
+            e.ctrlKey,
+            e.metaKey,
+            e.shiftKey,
+            movementPosition,
+            PointerId(e.pointerId),
+            e.width(magnification),
+            e.height(magnification),
+            e.pressure,
+            e.tangentialPressure,
+            Radians.fromDegrees(e.tiltX),
+            Radians.fromDegrees(e.tiltY),
+            Radians.fromDegrees(e.twist),
+            pointerType,
+            e.isPrimary
+          )
         )
+
+        if pointerType == PointerType.Mouse then {
+          globalEventStream.pushGlobalEvent(
+            MouseEvent.Leave(
+              position,
+              buttons,
+              e.altKey,
+              e.ctrlKey,
+              e.metaKey,
+              e.shiftKey,
+              movementPosition
+            )
+          )
+        }
       },
       onPointerDown = { e =>
-        val position = e.position(magnification, canvas)
+        val position         = e.position(magnification, canvas)
+        val buttons          = e.indigoButtons
+        val movementPosition = e.movementPosition(magnification)
+        val pointerType      = e.indigoPointerType
 
         globalEventStream.pushGlobalEvent(
-          PointerDown(position, PointerId(e.pointerId), Buttons(e.buttons), e.isPrimary)
+          PointerDown(
+            position,
+            buttons,
+            e.altKey,
+            e.ctrlKey,
+            e.metaKey,
+            e.shiftKey,
+            movementPosition,
+            PointerId(e.pointerId),
+            e.width(magnification),
+            e.height(magnification),
+            e.pressure,
+            e.tangentialPressure,
+            Radians.fromDegrees(e.tiltX),
+            Radians.fromDegrees(e.tiltY),
+            Radians.fromDegrees(e.twist),
+            pointerType,
+            e.isPrimary,
+            MouseButton.fromOrdinalOpt(e.button)
+          )
         )
-        MouseButton.fromOrdinalOpt(e.button).foreach { button =>
-          globalEventStream.pushGlobalEvent(MouseEvent.MouseDown(position, button))
+
+        if pointerType == PointerType.Mouse then {
```

Comment:
I'm not sure we want that do we? I was under the impression that pointer events were also used for things like touch input?

Reply:
Yes they are, but we should really only fire things like `MouseDown` if the pointer is actually a mouse. To be honest this is a deviation from the browser, which will fire a `MouseDown` even if what actually occured was a touch event. Happy to keep the more standard browser behaviour of course, I personally feel this is more 'correct', but there's an argument I think for both approaches